src/utils.jl

```diff
@@ -195,6 +195,15 @@ end
 @inline _gate(x::AbstractVector, h::Int, n::Int) = view(x, _gate(h, n))
 @inline _gate(x::AbstractMatrix, h::Int, n::Int) = view(x, _gate(h, n), :)
 
+@inline function _init_hidden_state(rng::AbstractRNG, rnn, x::AbstractMatrix)
+    return rnn.init_state(rng, rnn.out_dims, size(x, 2))
+end
+
+@inline function _init_hidden_state(rng::AbstractRNG, rnn,
+                                    x::Union{CUDA.StridedSubCuArray, CuArray})
+    return rnn.init_state(rng, rnn.out_dims, size(x, 2)) |> gpu
```

Comment:
I agree lets change this to 
```suggestion
    return CuArray(rnn.init_state(rng, rnn.out_dims, size(x, 2)))
```